src/mscorlib/src/System/Diagnostics/Contracts/ContractsBCL.cs

```diff
@@ -185,10 +185,15 @@ public void SetUnwind()
         }
     }
 
-    [SuppressMessage("Microsoft.Design", "CA1064:ExceptionsShouldBePublic")]
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    internal sealed class ContractException : Exception
+#if CORECLR
+    [SuppressMessage("Microsoft.Design", "CA1064:ExceptionsShouldBePublic")]
+    internal
+#else
+    public // On CoreRT this must be public.
```

Comment:
Include why it must be public in the comment?

Reply:
I wanted to match other place where we follow the same pattern but you are right, this should contain an explanation message. thanks